src/scipp/coords/transform_coords.py

```diff
@@ -82,27 +82,24 @@ def transform_coords(x: Union[DataArray, Dataset],
                       keep_intermediate=keep_intermediate,
                       keep_inputs=keep_inputs,
                       quiet=quiet)
-    if targets is None:
-        targets = []
-    if graph is None:
-        graph = {}
-
-    if isinstance(targets, dict):
-        if graph:
+    for field in fields(options):
+        if not isinstance(getattr(options, field.name), bool):
             raise ValueError(
-                "Got multiple transformation graphs, must provide only one.")
-        graph = targets
-        targets = set(targets)
-    else:
-        targets = {targets} if isinstance(targets, str) else set(targets)
+                f"'{field.name}' is a keyword reserved for options. "
+                "Use explicit targets and graph arguments to create an output "
+                "coordinate of this name.")
```

Comment:
'reserved for options' sounds strange. How about simply `'{field.name}' is a reserved keyword argument.`?